src/coloring/acyclic_coloring.jl

```diff
@@ -0,0 +1,201 @@
+                    w::Integer,
+                    x::Integer,
+                    g::LightGraphs.AbstractGraph,
+                    color::AbstractVector{<:Integer},
+                    forbidden_colors::AbstractVector{<:Integer},
+                    first_visit_to_tree::Array{Tuple{Integer, Integer}, 1},
+                    set::DisjointSets{LightGraphs.Edge})
+
+Subroutine to avoid generation of 2-colored cycle due to coloring of vertex v,
+which is adjacent to vertices w and x in graph g. Disjoint set is used to store
+the induced 2-colored subgraphs/trees where the id of set is a key edge of g
+"""
+function prevent_cycle(v::Integer,
+                        w::Integer,
+                        x::Integer,
+                        g::LightGraphs.AbstractGraph,
+                        color::AbstractVector{<:Integer},
+                        forbidden_colors::AbstractVector{<:Integer},
+                        first_visit_to_tree::AbstractArray{<: Tuple{Integer, Integer}, 1},
+                        set::DisjointSets{LightGraphs.Edge})
+
+    edge = find_edge(g, w, x)
+    e = find_root(set, edge)
+    p, q = first_visit_to_tree[edge_index(g, e)]
+    if p != v
+        first_visit_to_tree[edge_index(g, e)] = (v, w)
+    elseif q != w
+        forbidden_colors[color[x]] = v
+    end
+end
+
+"""
+        min_index(forbidden_colors::AbstractVector{<:Integer}, v::Integer)
+
+Returns min{i > 0 such that forbidden_colors[i] != v}
+"""
+function min_index(forbidden_colors::AbstractVector{<:Integer}, v::Integer)
+    return findfirst(!isequal(v), forbidden_colors)
+end
+
+"""
+        grow_star!(v::Integer,
+                w::Integer,
+                g::LightGraphs.AbstractGraph
+                first_neighbor::Array{Tuple{Integer, Integer}, 1},
+                set::DisjointSets{LightGraphs.Edge})
+
+Subroutine to grow a 2-colored star after assigning a new color to the
+previously uncolored vertex v, by comparing it with the adjacent vertex w.
+Disjoint set is used to store stars in sets, which are identified through key
+edges present in g.
+"""
+function grow_star!(set::DisjointSets{LightGraphs.Edge},
+                   v::Integer,
+                   w::Integer,
+                   g::LightGraphs.AbstractGraph,
+                   first_neighbor::AbstractArray{<: Tuple{Integer, Integer}, 1},
+                   color::AbstractVector{<: Integer})
+    edge = find_edge(g, v, w)
+    push!(set, edge)
+    p, q = first_neighbor[color[w]]
+    if p != v
+        first_neighbor[color[w]] = (v, w)
+    else
+        edge1 = find_edge(g, v, w)
+        edge2 = find_edge(g, p, q)
+        e1 = find_root(set, edge1)
+        e2 = find_root(set, edge2)
+        union!(set, e1, e2)
+    end
+    return nothing
+end
+
+
+"""
+        merge_trees!(v::Integer,
+                w::Integer,
+                x::Integer,
+                g::LightGraphs.AbstractGraph,
+                set::DisjointSets{LightGraphs.Edge})
```

Comment:
Signature in the doc seems to mismatch the actual signature below

Reply:
Fixed! Thanks.

Reply:
`set` comes first in the arguments of merge_trees!, but not in the documentation